### DISCUSSION TITLE
Enterprise level runner-scale-set on Github Enterprise Server

### DISCUSSION THREAD

**Discussion by 1MaxKoval** [actions/actions-runner-controller#3071] [FAQs] [answered]
Hello :wave:,

As the title suggests, is it possible to create a runner set at the enterprise level using the [GitHub Enterprise Server](https://docs.github.com/en/enterprise-server@3.6/admin/overview/about-github-enterprise-server) so that every organization and its repositories can utilize this runner set?

I noticed that the [documentation](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/deploying-runner-scale-sets-with-actions-runner-controller#choosing-runner-destinations) and [values.yaml](https://github.com/actions/actions-runner-controller/blob/1ae5d2b18e132a87033cbf34ae10b85b54c55a84/charts/gha-runner-scale-set/values.yaml#L3) only provide examples of creating runners at the enterprise level for `github.com`. There is no indication of whether such a runner set can exist at the enterprise level with GitHub Enterprise Server.

If it is possible, could you please provide an example of the appropriate setting for `githubConfigUrl`? 

If not (as of now), is this feature under consideration for future development?

So far, I have tried the following configurations for the `githubConfigUrl` value, none of which worked:
- `https://<self-hosted-name>`
- `https://<self-hosted-name>/`
- `https://<self-hosted-name>/<enterprise-name>`

When configured with any of the above values, the controller pod logs show a similar error:

```
2023-11-14T12:49:36Z    ERROR   Reconciler error        {"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com",
 "controllerKind": "AutoscalingRunnerSet", "AutoscalingRunnerSet": {"name":"entp-runner-set","namespace":"entp-arc-runners"}, 
"namespace": "entp-arc-runners", "name": "entp-runner-set", "reconcileID": "eb85dafb-8332-45b5-961d-d4d9717ecc7c", "error": "failed to 
parse githubConfigURL: \"https://<self-hosted-name>\": invalid config URL, should point to an enterprise, org, or repository"}
```

P.S.: It works fine with repository-level and organization-level runner creation when I configure the parameter respectively as:
- `https://<self-hosted-name>/<org-name>`
- `https://<self-hosted-name>/<org-name>/<repo-name>`

EDIT:

I just tried `https://<self-hosted-name>/enterprises/<local-enterprise-name>`, and it worked! I can see the runner available in all organizations/repos and is accepting jobs too :heart: !

My follow-up question is:

Although it works and seems fine, is this an expected argument value that will be supported later on (as it was not documented)?

**Comment by joshjohanning** [comment, marked as answer]
@1MaxKoval I think `/enterprises/enterprise-name` looks right, that's the URL I would expect to us there.

This also matches the API docs: https://docs.github.com/en/enterprise-server@3.10/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-an-enterprise

```bash
curl http(s)://HOSTNAME/api/v3/enterprises/ENTERPRISE/actions/runners/registration-token
```

Basically it is just appending `/actions/runners/registration-token` to the URL, as [you can see the same](https://docs.github.com/en/enterprise-server@3.10/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-an-organization) format with an org:

```bash
curl http(s)://HOSTNAME/api/v3/orgs/ORG/actions/runners/registration-token
```
